spatialprofilingtoolbox/db/credentials.py

```diff
@@ -0,0 +1,57 @@
+"""Structures and accessors for database credentials."""
+from os import environ
+import re
+import configparser
+
+from attr import define
+
+from spatialprofilingtoolbox.standalone_utilities.log_formats import colorized_logger
+
+logger = colorized_logger(__name__)
+
+@define
+class DBCredentials: #pylint: disable=too-few-public-methods
+    """Data structure for database credentials."""
+    endpoint: str
+    database: str
+    user: str
+    password: str
```

Comment:
Considered using a `NamedTuple` instead of a class?

Reply:
I noticed you used pydantics `BaseModel` elsewhere.

Reply:
Yeah the use of pydantic is strictly to work well with FastAPI. After some investigation into the options I generally prefer `attrs` for typing support. But since there are only 2 places I'm using `attrs` maybe the usage of `attrs` should be removed?
Yeah `NamedTuple` seems appropriate here.

Reply:
For now I'm keeping `attrs` in order to make use of its functionality of seamlessly providing `__init__` that does the right thing with arguments (for `SimpleReadOnlyProvider` subclasses). Otherwise we would have to add 5 new `__init__` methods with boilerplate.

I'll still change `DBCredentials` to `NamedTuple` though.

Reply:
I think that `NamedTuple` still has to be an explicitly defined `class` though, in order to get proper type checking.